src/player.rs

```diff
@@ -34,6 +40,29 @@ impl Player {
             steam_info: None,
             custom_data: serde_json::Value::Object(Map::new()),
             tags: Vec::new(),
+            local_verdict: Verdict::Player,
+            convicted: false,
+        }
+    }
+
+    /// Given a record, update this player with the included data.
+    pub fn update_from_record(&mut self, record: PlayerRecord) {
+        if record.steamid != self.steamid {
+            tracing::error!("Updating player with wrong record.");
+            return;
+        }
+
+        self.custom_data = record.custom_data;
+        self.local_verdict = record.verdict;
+    }
```

Comment:
Not exactly, the intention is that control flow should never be followed, and I was considering even putting a panic in there, however I would rather not introduce more ways for the program to crash and opted to just log an error there. Ideally it should probably be an assertion that will crash in debug mode for testing purposes but won't crash when running in release in case a common user is running the application. It is a little strange though I do acknowledge.